### PR TITLE
Integrate SC in Bitbucket pipeline

### PR DESCRIPTION
**Pull request by amelie-renard-sonarsource**

### CODE REVIEW

File: README.adoc [hunk 1]
```diff
@@ -1,3 +1,24 @@
+= C++ example project scanned on SonarCloud using Bitbucket Cloud pipelines
+// TODO: add SC status
+// TODO: add link to SC
+
+It is very easy to analyze a C/C++/Objective-C project on Bitbucket with SonarCloud:
+
+. Create a `sonar-project.properties` file to store your configuration
+. In your `bitbucket-pipelines.yml` file:
+.. Activate the https://docs.travis-ci.com/user/sonarcloud/[Travis Add-on]
```
Comment (arseniy-sonar) [inline README.adoc:9]: Is this travis addon still relevant to bitbucket?

File: README.adoc [hunk 2]
```diff
@@ -1,3 +1,24 @@
...
+.. Wrap your compilation with the Build Wrapper
+.. Run `sonar-scanner` later on
+
+You can take a look at the link:sonar-project.properties[sonar-project.properties] and link:.bitbucket-pipelines.yml[bitbucket-pipelines.yml] to see it in practice.
```
Comment (arseniy-sonar) [inline README.adoc:13]:
```suggestion
You can take a look at the link:sonar-project.properties[sonar-project.properties] and link:bitbucket-pipelines.yml[bitbucket-pipelines.yml] to see it in practice.
```

File: README.adoc [hunk 3]
```diff
@@ -1,3 +1,24 @@
...
+// TODO: add SC status
+// TODO: add link to SC
+
+It is very easy to analyze a C/C++/Objective-C project on Bitbucket with SonarCloud:
+
+. Create a `sonar-project.properties` file to store your configuration
+. In your `bitbucket-pipelines.yml` file:
+.. Activate the https://docs.travis-ci.com/user/sonarcloud/[Travis Add-on]
+.. Wrap your compilation with the Build Wrapper
+.. Run `sonar-scanner` later on
```
Comment (arseniy-sonar) [inline README.adoc:11]:
```suggestion
.. Run `sonar-scanner`
```

File: bitbucket-pipelines.yml [hunk 4]
```diff
@@ -14,7 +14,8 @@ definitions:
         script:
```
Comment (arseniy-sonar) [inline bitbucket-pipelines.yml:14]: Wouldn't it be nicer to break down this script into several steps, like we did in other CIs?

File: bitbucket-pipelines.yml [hunk 5]
```diff
@@ -14,7 +14,8 @@ definitions:
...
           - autoreconf --install
           - ./configure
-          - export SONAR_SCANNER_VERSION=4.4.0.2170
+          - export SONAR_SCANNER_VERSION=4.4.0.2170 # Find the latest version in the "Linux" link on this page:
+                                                    # https://sonarcloud.io/documentation/analysis/scan/sonarscanner/
           - export SONAR_SCANNER_HOME=$HOME/.sonar/sonar-scanner-$SONAR_SCANNER_VERSION-linux
           - export BW_OUTPUT=$HOME/.sonar/bw-output
```
Comment (arseniy-sonar) [inline bitbucket-pipelines.yml:20]: Is there a way to extract the environment variable definitions outside of the script into a separate yml block, like [in gh-actions](https://github.com/sc-sq-c-family-examples/macos-xcode-gh-actions-sc/blob/main/.github/workflows/build.yml#L12)?

File: bitbucket-pipelines.yml (hunk 4 above)
Comment (arseniy-sonar) [reply bitbucket-pipelines.yml:14]: I would also fetch and install the dependencies, sonar-scanner and build-wrapper, first, before running the build, to mirror the scripts for other CIs.

**Review by arseniy-sonar — CHANGES_REQUESTED** [review]

File: README.adoc [hunk 6]
```diff
@@ -1,3 +1,25 @@
...
+. Add a repository variable `SONAR_TOKEN` with the value given by SonarCloud during the setup
+. In your `bitbucket-pipelines.yml` file:
+.. Download the https://sonarcloud.io/documentation/analysis/languages/cfamily/#analysis-steps-using-build-wrapper[`build wrapper`] and wrap your compilation with it
+.. Download and launch https://sonarcloud.io/documentation/analysis/scan/sonarscanner/[`sonar-scanner`]
+
+You can take a look at the link:sonar-project.properties[sonar-project.properties] and link:bitbucket-pipelines.yml[bitbucket-pipelines.yml] to see it in practice.
+
+= Documentation
+
+- https://docs.sonarqube.org/latest/analysis/languages/cfamily/[Documentation of the C/C++/Objective-C plugin and its Bsuild Wrapper]
```
Comment (arseniy-sonar) [inline README.adoc:19]:
```suggestion
- https://docs.sonarqube.org/latest/analysis/languages/cfamily/[Documentation of the C/C++/Objective-C plugin and its Build Wrapper]
```

**Review by amelie-renard-sonarsource — COMMENTED** [review]
Ok, that's weird... I replied to your comments yesterday and they appear as "pending" today, as if I did a review... Can't answer directly without finishing the review

File: bitbucket-pipelines.yml (hunk 4 above)
Comment (arseniy-sonar) [reply bitbucket-pipelines.yml:14]: Alright, I think it is not worth the complexity.

File: README.adoc [hunk 7]
```diff
@@ -1,3 +1,25 @@
+= C++ example project scanned on SonarCloud using Bitbucket Cloud pipelines
+// TODO: add SC status
+// TODO: add link to SC
```
Comment (arseniy-sonar) [inline README.adoc:3]: These links

File: README.adoc [hunk 8]
```diff
@@ -1,3 +1,25 @@
...
+
+To analyze a C/C++/Objective-C project on Bitbucket pipelines with SonarCloud, follow the steps described here: https://sonarcloud.io/documentation/getting-started/bitbucket-cloud/[Getting started with Bitbucket Cloud].
+
+To sum up, you need to:
+
+. Enable pipelines in your Bitbucket Cloud repository
```
Comment (arseniy-sonar) [inline README.adoc:9]: Perhaps, keep the step about filling sonar-project.properties, as in other examples.

File: README.adoc [hunk 9]
```diff
@@ -1,3 +1,25 @@
...
+.. Download the https://sonarcloud.io/documentation/analysis/languages/cfamily/#analysis-steps-using-build-wrapper[`build wrapper`] and wrap your compilation with it
+.. Download and launch https://sonarcloud.io/documentation/analysis/scan/sonarscanner/[`sonar-scanner`]
+
+You can take a look at the link:sonar-project.properties[sonar-project.properties] and link:bitbucket-pipelines.yml[bitbucket-pipelines.yml] to see it in practice.
+
+= Documentation
+
+- https://docs.sonarqube.org/latest/analysis/languages/cfamily/[Documentation of the C/C++/Objective-C plugin and its Build Wrapper]
+- https://sonarcloud.io/documentation/reference/bitbucket-integration/[Bitbucket Cloud Integration]
+- https://sonarcloud.io/documentation/reference/ci/bitbucket-pipelines/[Analyze your repository with Bitbucket Pipelines]
```
Comment (arseniy-sonar) [inline README.adoc:21]: It is not clear what is the difference between these two links.

**Review by arseniy-sonar — CHANGES_REQUESTED** [review]
Just a couple of fixes

File: README.adoc (hunk 7 above)
Comment (amelie-renard-sonarsource) [reply README.adoc:3]: I wait for the repo to be mirrored on bitbucket and then I'll create all the links, checking that all is correctly linked

**Review by arseniy-sonar — APPROVED** [review]

**Pull request merged**